monash-human-power/MHP_DAS
# Add systemd file for Mqtt logger

## Description

This PR adds the files needed to set up our own systemd service for the MQTT logger. It also makes a change to the `V3_fake_module.py` script to send a stop message whenever the script is terminated.

## Screenshots

After running `install.sh` you should see the `logger` in the user systemd folder:
<img width="667" alt="Screen Shot 2021-08-30 at 12 07 22 pm" src="https://user-images.githubusercontent.com/63642262/131291692-6e40352c-2f8a-4b74-98b7-39153ee4428f.png">

Logs from the `logger` service when data is being published and then stopped:
_Note: The printing of `/v3/wireless_module/3/data` was manually added just to check the logger was working correctly:_

<img width="851" alt="Screen Shot 2021-08-30 at 12 06 47 pm" src="https://user-images.githubusercontent.com/63642262/131291678-c3ff8844-d535-44a7-8d89-b1b69c58f5ef.png">



## Steps to Test

1) On your machine or Raspberry Pi, on the command line run `./install.sh` to set up the `logger` service
2) Check the systemd folder for user service files to confirm logger service is present there
3) Follow the README to enable and then start the service
4) Run the fake WM script and then stop it to see if the logger has successfully logged the data and create a csv file in the `das` folder.
